scipy/stats/morestats.py

```diff
@@ -1222,12 +1222,33 @@ def shapiro(x, a=None, reta=False):
     See Also
     --------
     anderson : The Anderson-Darling test for normality
+    kstest : The Kolmogorov-Smirnov test for goodness of fit.
 
     References
     ----------
     .. [1] http://www.itl.nist.gov/div898/handbook/prc/section2/prc213.htm
+    .. [2] Shapiro, S. S. & Wilk, M.B (1965). An analysis of variance test for
+           normality (complete samples), Biometrika, Vol. 52, pp. 591-611.
+    .. [3] Razali, N. M. & Wah, Y. B. (2011) Power comparisons of Shapiro-Wilk,
+           Kolmogorov-Smirnov, Lilliefors and Anderson-Darling tests, Journal of
+           Statistical Modeling and Analytics, Vol. 2, pp. 21-33.
+    .. [4] ALGORITHM AS R94 APPL. STATIST. (1995) VOL. 44, NO. 4.
+
+    Examples
+    --------
+    >>> from scipy import stats
+    >>> np.random.seed(12345678)
+    >>> x = stats.norm.rvs(loc=5, scale=3, size=100)
+    >>> stats.shapiro(x)
+    (0.9772805571556091, 0.08144091814756393)
 
     """
+    x = np.ravel(x)
+    if x.dtype == object:
+        # np.ravel and np.ndarray.flatten don't work with dtype=object
+        # i.e. they will fail to flatten if there are different sized arrays.
+        x = np.hstack(x)
```

Comment:
It doesn't actually make it silently return wrong values like I had thought, it just errors. If this is considered an acceptable result I can remove the added code.

e.g.
```
stats.shapiro([[1, 2], [3, 4], [5]])
```

With the above code:
```
(0.9867621660232544, 0.9671739339828491)
```
(Same as `stats.shapiro([1, 2, 3, 4, 5])` which is correct)

Without the aboove code:
```
ValueError: setting an array element with a sequence.
```